### ISSUE TITLE
ComplianceTest_Toolkit: Wrong correction message given when dlls in the same solution is used

### ISSUE THREAD

**Issue by IsakNaslundBh** [BHoM/Test_Toolkit#246]
#### Description:

As can be seen here:

https://github.com/BHoM/BHoM_UI/runs/655697979

References to dlls in a project in the same solution are correctly flagged as being wrong, but the solution text is wrong. Should not be stating that you need to go back in relative path and into the BHoM_UI folder (for this case) but should state that this should be referenced as a project, and not as a dll.

Good to get fixed for clarity in messaging.

Basically, references to a a dll outside the solution are currently correct
Reference to a dll inside the solution is currently wrong in terms of solution message

**Comment by FraserGreenroyd** [comment]
This would technically be a brand new check required @IsakNaslundBh , as currently there is no compliance that forces you to use `by-project` reference in the wiki, though we do of course strongly encourage it for obvious reasons.

The next question is how would you programmatically achieve this, keeping in mind the complexity of BHoM and BHoM_Engine with multiple projects and reference names that are shared across two repos, one which would need to be set to a local path (for the oM) and one for a project reference (which is currently never checked anyhow).

Alternatively, the suggested path was originally designed as just that, a suggestion of where we think it ought to be found. We can change the message to better imply that it is a suggestion and not an absolute instead, but given the suggestion usually comes from where it expects to find it this might be counter-intuitive cause if you do decide not to follow the suggestion, the CI will still report it as a failure the next time around. So an option on the table but not one I'd recommend.

**Comment by IsakNaslundBh** [comment]
Was mostly in relation to this:

![image](https://user-images.githubusercontent.com/22005920/81400157-c7821100-912c-11ea-8261-ff04195d5dc4.png)

Where the flagging as it being wrong is correct, but the solution being wrong. Don't think you even would be able to get the path suggested if you are in the same solution file, which also is confusing.

So not sure if a brand new check is fully needed, as the current one actually are flagging them up as wrong, or if we just need to improve the messaging.

Ofc, a brand new check explicitly checking that and projects inside the solution is referenced as a project, and not a dll, would be even more ideal.

and

> This would technically be a brand new check required @IsakNaslundBh , as currently there is no compliance that forces you to use by-project reference in the wiki, though we do of course strongly encourage it for obvious reasons.

the current check will flag this up, but as shown in the link, which to a large extent is even more confusing.
